src/audio/module_adapter/module/modules.c

```diff
@@ -136,7 +136,7 @@ static int modules_init(struct processing_module *mod)
 	}
 	comp_info(dev, "modules_init() start");
 
-	if (!md->module_adapter && md->ops == &interface) {
+	if (!md->llext || md->ops == &interface) {
```

Comment:
The `module_data` structure is unique for each module instance. You will not be able to tell whether this is the first loading of the module based on the value of any of its fields. I suggest following solution: https://github.com/thesofproject/sof/pull/8939